src/main/java/org/mamute/mail/NewQuestionMailer.java

```diff
@@ -0,0 +1,51 @@
+package org.mamute.mail;
+
+import br.com.caelum.vraptor.environment.Property;
+import br.com.caelum.vraptor.simplemail.AsyncMailer;
+import br.com.caelum.vraptor.simplemail.Mailer;
+import br.com.caelum.vraptor.simplemail.template.BundleFormatter;
+import br.com.caelum.vraptor.simplemail.template.TemplateMail;
+import br.com.caelum.vraptor.simplemail.template.TemplateMailer;
+import br.com.caelum.vraptor.view.LinkToHandler;
+import org.apache.commons.mail.Email;
+import org.apache.commons.mail.EmailException;
+import org.apache.log4j.Logger;
+import org.mamute.controllers.QuestionController;
+import org.mamute.model.Question;
+import org.mamute.model.User;
+import org.mamute.vraptor.Linker;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class NewQuestionMailer {
+
+	@Inject
+	private AsyncMailer mailer;
+	@Inject
+	private TemplateMailer templates;
+	@Inject
+	private BundleFormatter bundle;
+	@Inject
+	@Property("mail_logo_url")
+	private String emailLogo;
+	@Inject
+	private Linker linker;
+
+	public void send(List<User> subscribed, Question question) {
+		linker.linkTo(QuestionController.class).showQuestion(question, question.getSluggedTitle());
+		String questionLink = linker.get();
+		TemplateMail template = templates.template("new_question_notification")
+				.with("question", question)
+				.with("bundle", bundle)
+				.with("questionLink", questionLink)
+				.with("logoUrl", emailLogo);
+		for (User user : subscribed) {
+			boolean notSameAuthor = !user.equals(question.getAuthor());
+			if (notSameAuthor) {
+				Email email = template.to(user.getName(), user.getEmail());
+				mailer.asyncSend(email);
+			}
```

Comment:
it's really hard to test it, I'd have to create a bunch of mocks. I can also create an guava's predicate and use it here with Collections2.filter and unit test the predicate, what do you think? Is that too much just for a unit test?